Feature/activity search url

pe-st opened the pull request:
Use new endpoint for getting all activities, see the explanation in README.md

yohcop committed:
Add JSON export format.

JSON format does not download any file, but saves all the metadata received about activities in a .json file.

julienr committed:
Fix download URLs to point to modern proxy so TCX/GPX download works.
Use HTTPS for download

pe-st committed:
Merge pull request #2 from julienr/fix_download_urls

Fix download URLs to point to modern proxy so TCX/GPX download works.

pe-st committed:
Write an empty GPX file when activity has no GPS data and returns HTTP 204

pe-st committed:
Merge remote-tracking branch 'origin/http-204' into develop

pe-st committed:
Merge and adapt yohcop's pull request (pr/6) into develop

pe-st committed:
Start using a different endpoint for the list

The old endpoint ('activity-search-service-1.2') doesn't seem to
work anymore for more than 20 activities, but is still useful for
the result summary (total numbers of activities)

The new endpoint ('activitylist-service') has an almost
completely different JSON output, which hasn't been adapted yet.

pe-st committed:
Convert 4 more columns to new JSON format

pe-st committed:
Convert duration, distance ans speed columns

Also explain limit_maximum

pe-st committed:
Add uncorrected elevation columns

pe-st committed:
Convert more columns to new JSON format

pe-st committed:
Improve timestamps

pe-st committed:
Convert 3 more columns to new JSON format

pe-st committed:
Update README.md

pe-st committed:
Convert the remaining columns to new JSON format

pe-st committed:
Fix behaviour for some spurious null fields

pe-st committed:
Replace develop branch with current master

pe-st committed:
Merge 'feature/activity-search-url' into develop

moderation commented:
This is a pretty huge change and modifies how I would use the utility and the data inside of it. The current CSV output I use goes into an Google Spreadsheet that dates back many years. I'm tied up for the next couple of weeks but will review then to see what makes sense and possible suggest splitting this PR up into small bits.  Thanks @pe-st

pe-st commented:
No problem here. Which columns do bother you?
I tried to keep the CSV output mostly as it was (except maybe for the timestamps where I tried to have a more standard format - the ISO format), concentrating on making the script work again. It's not yet the format I would prefer (my spreadsheet is based on the original repo from kjkjava, but now there are more data available like VO2max I'd like to integrate into my spreadsheet), but my personal changes I'll do on a private branch.

pe-st committed:
Restore Activity ID and timestamp order

pe-st committed:
Restore order for 4/5 more columns

pe-st committed:
Restore Max Elevation (corrected or not)

pe-st committed:
Add JSON export format.

This an adaptation of commit 7b0fcc76 to the refactored code

pe-st committed:
Merge feature/json-export into current branch

pe-st committed:
Retrieve device info

pe-st committed:
Merge 'moderation_origin/master' into develop

pe-st committed:
Merge branch 'develop' into current branch

pe-st committed:
Improve error handling for device service

pe-st committed:
Merge feature/json-export into develop

pe-st committed:
Retrieve device info

Cherry-Pick of commits 234c2296 and a61dee18

pe-st committed:
Merge branch 'develop' into current branch

pe-st committed:
Restore more, use different timestamp format

 - restored order for a dozen or so columns
 - timestamp format is almost RFC1123 as used by Garmin

pe-st committed:
Restore remaining columns

pe-st committed:
Round endTimeStamp to seconds

Add activity detail example file corresponding to the other examples

pe-st committed:
Recreate old endTime

pe-st committed:
Add mapping for the parentTypeId

pe-st committed:
Add device version, adjust displayed precision

pe-st committed:
Truncate decimal degrees to 6 decimal places

6 decimal places corresponds to a precision of 4 to 11 cm, way enough

pe-st committed:
Output pace/speed depending on activity type

pe-st committed:
Fix rounding of duration

pe-st committed:
Revert to rounding the duration

pe-st committed:
Use elevation data from detail JSON

pe-st committed:
Use elapsed, use more data from detail JSON

pe-st committed:
Use fallback for longitude/latitude

pe-st committed:
Retry the endpoint call up to 3 times

pe-st committed:
Recreate old output for empty name/device

pe-st committed:
Remove superfluous last comma in CSV lines

Cleanup also other stuff like unneeded quote escaping for columns
that will never see a quote, etc.

pe-st committed:
Use integers for heart rate

no point in using a float with one decimal place

pe-st committed:
Use calories from detail JSON

pe-st committed:
Use an almost RFC1123 timestamp format

This format is almost the same that Garmin used
in the activity-search-service

pe-st committed:
Add device version, change some precisions

pe-st committed:
Output pace/speed depending on activity type

pe-st committed:
Use elevations, lati-/longitude etc from detail

pe-st committed:
Retry the endpoint call up to 3 times

pe-st committed:
Remove superfluous last comma in CSV lines

Cleanup also other stuff like unneeded quote escaping for columns
that will never see a quote, etc.

pe-st committed:
Use integers for HR, calories from detail

no point in using a float with one decimal place for heart rate

pe-st committed:
Change the login to Garmin Connect

The ticket is now extracted from the login response.
The cookies are not needed anymore, as well as the
voodoo calls to modern and legacy session.

For the post_auth call I chose just an URL that worked when
analyzing the HTTP traffic in the browser, maybe there are better
ones (with less payload)?

Issue moderation/garmin-connect-export#10

pe-st committed:
Merge 'feature/change-login' into develop

Issue moderation/garmin-connect-export#10

pe-st committed:
Update README.md

pe-st committed:
Merge 'develop' into feature/activity-search-url

- Add separate REST call for every activity,
  restoring some info missing from original PR
- Also include code for moderation/garmin-connect-export#10

pe-st committed:
Remove unneeded checks for 'SummaryDTO'

It's already checked before

pe-st committed:
Use temperatures from detail

The detail contains also average and max temperature

pe-st committed:
Pretty-print activity and event types

The 'pretty' names come from a dictionary retrieved by separate
REST calls

pe-st committed:
Add a section about choices made

pe-st committed:
Restore formats for calories and timestamps

pe-st committed:
Merge branch 'develop' into current branch

pe-st commented:
Hello Michael @moderation 

I think I've adapted the script to give almost the same CSV that you got using the old endpoints.

Known differences:

- for some activity types the speed columns contain the pace instead of km/h. This can easily be changed by setting `uses_pace` to an empty set, though
- If I understand the few data that I managed to get with the old endpoint before it stopped working completely, your `xxx timestamp (ms)` contain a UTC timestamp in ISO format. My output has the local time with the offset (e.g. `+02:00`), I think it's more useful.
- The timezone column contains the correct offset instead of just a time zone name with potentially wrong offset (e.g. `(GMT+01:00) Central European Time` is wrong during daylight savings time...)
- I haven't found an endpoint giving both corrected and uncorrected elevations. The script fills just the columns provided by the activity and leaves the other 4 columns empty
- latitudes and longitude are cut to 6 decimal places, which corresponds to a precision of about 10 cm (https://en.wikipedia.org/wiki/Decimal_degrees), which is probably still a far better precision than current devices can achieve. The displayed precision can easily be changed.
- The minimum heart rate doesn't seem to exist anymore
- I have no data with WeightedMeanBikeCadence, MaxBikeCadence, SumStrokes or SumSampleCountDuration, so I dont know if they are correct
